src/app/_helpers/calcprogress.ts

```diff
@@ -0,0 +1,13 @@
+import { WorkflowExecution } from '../_models';
+
+export function calcProgress(ongoing: WorkflowExecution): number {
+  if (!ongoing.currentPluginIndex) {
```

Comment:
if the currentPluginx === 0, this condition will be triggered and 0 will be returned. But that is not what you want, you still want to calculate the subPct and return that. So you should do something like `if (ongoing.currentPluginIndex !== undefined)`.